### ISSUE TITLE
Remote user guard: 500 error about type conversion

### ISSUE THREAD

**Issue by hakoerber** [firefly-iii/firefly-iii#4060]
**Bug description**
Firefly Version: 5.4.6

I am using the remote user authentication, which worked perfectly in 5.4.5. Since updating to 5.4.6, I'm getting the following error:

> Array to string conversion
> 
> This error occured in file `/var/www/html/app/Support/Authentication/RemoteUserGuard.php` on line 83 with code 0.

**Steps to reproduce**
* Use `AUTHENTICATION_GUARD=remote_user_guard`
* Use `AUTHENTICATION_GUARD_HEADER=HTTP_X_AUTH_USERNAME` (I am not sure whether this affects this bug, if you suspect it does I can also change this)
* Access firefly

**Some more Information**

The reported line was changed [this commit](https://github.com/firefly-iii/firefly-iii/commit/647c9c5eb027440e29c6a3b86f731640df9c0d36), so I suspect this is related to the fix for #4023. 

I'm not sure why the value of the username header should be an array. I checked, and it's set as a string in my authentication reverse proxy.

**Comment by JC5** [comment]
Neither do I. Please add this line before the $emailAddress line:

```
Log::debug(sprintf('Header value: %s', var_export(request()->server($header), true)));
```

**Comment by hakoerber** [comment]
it prints a bunch of headers, looks like *all* headers (redacted):

```
[2020-11-14 18:04:55] local.NOTICE: Header value: array (
  'HTTP_HOST' => '127.0.0.1:8080',
  'HTTP_CONNECTION' => 'close',
[...]
)
```

I debugged a bit further and found `$header` to be `NULL` in [this line](https://github.com/firefly-iii/firefly-iii/blob/bcb734a4593e1f368b9e6cc4f3157cbc667d6d9d/app/Support/Authentication/RemoteUserGuard.php#L82). This leads to leads to the dumping of **all** headers. It's of course empty because I did not set the `auth.guard_email` config (via env variable `AUTHENTICATION_GUARD_EMAIL`).

When I explictly specify the header via `AUTHENTICATION_GUARD_EMAIL`, all is well.

Note that it's [empty by default](https://github.com/firefly-iii/firefly-iii/blob/bcb734a4593e1f368b9e6cc4f3157cbc667d6d9d/.env.example#L197), which might be quite confusing when using remote user guard authentication on a fresh setup. The functionality before [the change](https://github.com/firefly-iii/firefly-iii/commit/647c9c5eb027440e29c6a3b86f731640df9c0d36) just worked "by accident". 

I think it would be a good idea to do an explicit check whether `config('auth.guard_email')` is empty, and skip the email setup in this case.

If you want, I can open a pull request for that.
